src/app/state/rewardSlice.ts

```diff
@@ -83,53 +90,55 @@ type NonFungibleResource = NonFungibleResourcesCollectionItem & {
 // when fetching receipts for rewards, you need to fetch receipts for all pairs (not just the DEXTER/XRD pair).
 // You can use the existing alphadex api endpoint: /account/orders to get the orders for each pair.
 export const fetchReciepts = createAsyncThunk<
-  undefined, // Return type of the payload creator
+  string[], // Return type of the payload creator
   undefined, // argument type
   {
     state: RootState;
   }
->("rewards/fetchReciepts", async (_, thunkAPI) => {
-  const dispatch = thunkAPI.dispatch;
+>("rewards/fetchReciepts", async () => {
   const rdt = getRdt();
-  if (!rdt) return;
+  if (!rdt) {
+    throw new Error("RDT initialization failed");
+  }
+
+  // Question from @dcts: should those be really taken from .env
+  // or are we fetchnig those too with fetchAddresses ?
   const claimComponentAddress = process.env.NEXT_PUBLIC_CLAIM_COMPONENT;
   const resourceAddress = process.env.NEXT_PUBLIC_RESOURCE_ADDRESS_DEXTERXRD;
-  if (!claimComponentAddress) return;
+  if (!claimComponentAddress) {
+    throw new Error("claimComponentAddress missing");
+  }
+  if (!resourceAddress) {
+    throw new Error("resourceAddress missing");
+  }
   const walletData = rdt.walletApi.getWalletData();
-  //Todo support multiple wallets ids
+  // Todo support multiple wallets ids
   const accountAddress = walletData.accounts[0].address;
 
-  try {
-    const response =
-      await rdt.gatewayApi.state.innerClient.entityNonFungiblesPage({
-        stateEntityNonFungiblesPageRequest: {
-          address: accountAddress,
-          // eslint-disable-next-line camelcase
-          aggregation_level: "Vault",
-          // eslint-disable-next-line camelcase
-          opt_ins: { non_fungible_include_nfids: true },
-        },
-      });
-    const { items } = response;
-
-    const accountReceiptVault =
-      (items.find(
+  const { items } =
+    await rdt.gatewayApi.state.innerClient.entityNonFungiblesPage({
+      stateEntityNonFungiblesPageRequest: {
+        address: accountAddress,
         // eslint-disable-next-line camelcase
-        ({ resource_address }) => resource_address === resourceAddress
-      ) as NonFungibleResource) || null;
-
-    if (accountReceiptVault && accountReceiptVault?.vaults.items.length > 0) {
-      dispatch(
-        rewardSlice.actions.updateReciepts(
-          accountReceiptVault?.vaults.items[0].items as string[]
-        )
-      );
-    }
-  } catch (error) {
-    return undefined;
```

Comment:
I suggest to let the error being thrown, catching the error to return undefined will make this error not catchable.